sklearn/utils/_mask.py

```diff
@@ -1,19 +1,41 @@
 import numpy as np
+from scipy import sparse
 
 from . import is_scalar_nan
 from .fixes import _object_dtype_isnan
 
 
-def _get_mask(X, value_to_mask):
-    """Compute the boolean mask X == value_to_mask."""
+def _fit_mask(X, value_to_mask):
     if is_scalar_nan(value_to_mask):
         if X.dtype.kind == "f":
-            return np.isnan(X)
+            Xt = np.isnan(X)
         elif X.dtype.kind in ("i", "u"):
             # can't have NaNs in integer array.
-            return np.zeros(X.shape, dtype=bool)
+            Xt = np.zeros(X.shape, dtype=bool)
         else:
             # np.isnan does not work on object dtypes.
-            return _object_dtype_isnan(X)
+            Xt = _object_dtype_isnan(X)
     else:
-        return X == value_to_mask
+        Xt == value_to_mask
+    return Xt
+
+
+def _get_mask(X, value_to_mask, reconstruct_sparse=False):
+    """Compute the boolean mask X == missing_values."""
+    # We get entire sparse matrix when reconstruct is True
+    # Otherwise we get X.data if X is sparse, and X when X is dense
+    if sparse.issparse(X) and reconstruct_sparse:
+        Xt = _fit_mask(X.data, value_to_mask)
+    # if we need not reconstruct sparse or X is dense we can directly
```

Comment:
I would not put this comment. I think it might be easier to read the code in the opposite way

```python
if not (sparse.issparse(X) and reconstruct_sparse):
    return _fit_mask(X, value_to_mask)
Xt = _fit_mask(X.data, value_to_mask)
...
```

Reply:
Done

Reply:
Please let me know if any more changes are needed